worlds/terraria/Checks.py

```diff
@@ -0,0 +1,362 @@
+    "Old One's Army Tier 1",
+    "Queen Bee",
+    "Skeletron",
+    "Deerclops",
+    "Wall of Flesh",
+]
+
+post_wall_of_flesh_locations = [
+    "Pirate Invasion",
+    "Queen Slime",
+    "The Twins",
+    "The Destroyer",
+    "Skeletron Prime",
+    "Old One's Army Tier 2",
+    "Plantera",
+    "Duke Fishron",
+]
+
+post_plantera_locations = [
+    "Frost Legion",
+    "Golem",
+    "Old One's Army Tier 3",
+    "Martian Madness",
+    "Mourning Wood",
+    "Pumpking",
+    "Everscream",
+    "Santa-NK1",
+    "Ice Queen",
+    "Empress of Light",
+    "Lunatic Cultist",
+    "Lunar Events",
+    "Moon Lord",
+]
+
+post_moon_lord_locations = [
+    "Zenith",
+]
+
+achievements = [
+    "Timber!!",
+    "No Hobo",
+    "Stop! Hammer Time!",
+    "Ooo! Shiny!",
+    "Heart Breaker",
+    "Heavy Metal",
+    "I Am Loot!",
+    "Star Power",
+    "Hold on Tight!",
+    "Smashing, Poppet!",
+    "Where's My Honey?",
+    "Dungeon Heist",
+    "It's Getting Hot in Here",
+    "Miner for Fire",
+    "Like a Boss",
+    "Bloodbath",
+    "Not the Bees!",
+    "Jeepers Creepers",
+    "Funkytown",
+    "Into Orbit",
+    "Rock Bottom",
+    "Fashion Statement",
+    "Vehicular Manslaughter",
+    "Lucky Break",
+    "Throwing Lines",
+    "Dye Hard",
+    "The Cavalry",
+    "Completely Awesome",
+    "Til Death...",
+    "Watch Your Step!",
+    "You Can Do It!",
+    "Matching Attire",
+    "Benched",
+    "Quiet Neighborhood",
+    "Hot Reels!",
+    "Leading Landlord",
+    "Feeling Petty",
+]
+
+grindy_achievements = [
+    "Sticky Situation",
+    "There are Some Who Call Him...",
+    "Deceiver of Fools",
+    "Archaeologist",
+    "Pretty in Pink",
+    "Boots of the Hero",
+    "A Rather Blustery Day",
+    "Heliophobia",
+    "Jolly Jamboree",
+    "Dead Men Tell No Tales",
+]
+
+fishing_achievements = [
+    "Glorious Golden Pole",
+    "Servant-in-Training",
+    "10 Fishing Quests", # This achievement was intentionally renamed
+    "Trout Monkey",
+    "Fast and Fishious",
+    "Supreme Helper Minion!",
+]
+
+achievements_that_are_fishing_if_goal_is_wof = [
+    "Head in the Clouds",
+]
+
+post_wall_of_flesh_achievements = [
+    "Begone, Evil!",
+    "Extra Shiny!",
+    "Drax Attax",
+    "Photosynthesis",
+    "Get a Life",
+    "Kill the Sun",
+    "Mecha Mayhem",
+    "Prismancer",
+    "It Can Talk?!",
+]
+
+grindy_post_wall_of_flesh_achievements = [
+    "Gelatin World Tour",
+    "Topped Off",
+    "Don't Dread on Me",
+]
+
+post_plantera_achievements = [
+    "Temple Raider",
+    "Robbing the Grave",
+    "Baleful Harvest",
+    "Ice Scream",
+    "Sword of the Hero",
+]
+
+grindy_post_plantera_achievements = [
+    "Big Booty",
+    "Real Estate Agent",
+    "Rainbows and Unicorns",
+]
+
+post_moon_lord_achievements = [
+    "Sick Throw",
+    "You and What Army?",
+]
+
+next_id = 0x7E0000
+
+item_name_to_id: dict[str, int] = {}
+location_name_to_id: dict[str, int] = {}
+
+def register(names, isItem):
+    global next_id
```

Comment:
Is this usage of a global variable safe when multiple Terraria worlds are generated at once? I don't think it will cause any problems, but I'd feel safer knowing if it was explicitly tested.

Reply:
I couldn't get multiple worlds to run the same part of the code at once, so I couldn't perform a good test. Instead, I replaced it with a solution that doesn't involve global mutability